sw/device/silicon_creator/lib/chip_info.h

```diff
@@ -0,0 +1,25 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CHIP_INFO_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_CHIP_INFO_H_
+
+#include <assert.h>
+#include <stdint.h>
+
+typedef struct chip_info {
+  uint32_t scm_revision;
+  char build_date_str[20];
+} chip_info_t;
```

Comment:
@cfrantz please chime in if we should include anything else.

Reply:
Perhaps a enum constant representing that this version of the ROM is for earlgrey (as opposed to other projects in the works).

Do we want more than the top 32-bits of the git rev? (ie: expand this to uint64_t)

As in olden times, maybe the names of the people who made this thing? :)  Better yet, [a photo](https://www.cocopedia.com/wiki/index.php/CoCo3_Easter_Egg).

Reply:
> As in olden times, maybe the names of the people who made this thing? :) Better yet, [a photo](https://www.cocopedia.com/wiki/index.php/CoCo3_Easter_Egg).

Haha, great minds think alike, we all have great minds :)

Reply:
I think we should add the enum constant to make identifying easier in the field.

I'm on the fence about `uint64_t` because I think it is highly unlikely to have a collision in the first 32-bits and the colliding commit also being a ROM release. That being said, this would require some additional logic or manual intervention. We should probably add it.

Reply:
Sounds good. I'll increase the commit hash to 64 bits and I'll add an enum with `kEarlgrey` and `kEnglishbreakfast`.

Reply:
Actually, I opted to use the reset value of `lc_ctrl.HW_REV` instead of adding a "top" enum, per @alphan's advice.